### ISSUE TITLE
Fluent API Deadlocks

### ISSUE THREAD

**Issue by LukaszSzulc** [Azure/azure-sdk-for-net#3469]
In Console Application everything work's well

             IVirtualMachine vm = await azure.VirtualMachines.GetByResourceGroupAsync("*", "*");

             PowerState powerState = vm.PowerState;

             string ip = vm.GetPrimaryPublicIPAddress().IPAddress;`

but in Web Application I must wrap getting PowerState and IP inside new Task

`
            
            IVirtualMachine vm = await azure.VirtualMachines.GetByResourceGroupAsync("*", "*");

            PowerState powerState = null;

            string ip = null;

            await Task.Run(() => ip = vm.GetPrimaryPublicIPAddress().IPAddress);

            await Task.Run(() => powerState = vm.PowerState);
`

Without that application is unresponisive. This state is probably result of deadlock in GetPrimaryPublicIpAddress() method and PowerState property.

**Comment by anuchandy** [comment]
We need to implement `IVirtualMachine.GetPrimaryPublicIPAddressAsync()`.

For PowerState, can you try

```C#
VirtualMachineInstanceView instanceView = await vm.RefreshInstanceViewAsync();
PowerState powerState = Fluent.PowerState.FromInstanceView(instanceView);
```

**Comment by tugberkugurlu** [comment]
I get the same when running `List` on `IDatabases` which you get by listing the SQL Servers on Azure. This happens under ASP.NET which has an ambient `SynchronizationContext`. I sense that somewhere it has been skipped to await on `ConfigureAwait(false)` and List calls an async method in a blocking way.

**Comment by abergs** [comment]
I get this on List() on Elastpools and also on List Websites

**Comment by anuchandy** [comment]
@abergs, @tugberkugurlu ASP.NET dead lock issue is fixed in this PR https://github.com/Azure/azure-sdk-for-net/pull/3501. This fix will be available as part of the upcoming release (scheduled for next week).

**Comment by LukaszSzulc** [comment]
@anuchandy Thank you It will help me in my project.

**Comment by martinsawicki** [comment]
asp.net related deadlock issues should have been addressed in v1.2
